src/dbnode/encoding/ostream.go

```diff
@@ -155,10 +155,17 @@ func (os *ostream) WriteByte(v byte) {
 
 // WriteBytes writes a byte slice.
 func (os *ostream) WriteBytes(bytes []byte) {
+	if !os.hasUnusedBits() {
+		// If the stream is aligned on a byte boundary then all of the WriteByte()
+		// function calls and bit-twiddling can be skipped in favor of a single
+		// memcopy.
```

Comment:
You're right, what I'm doing would bypass the pooling. I'll change it to call ensureCapacityFor() first.

Reply:
I'm gonna stick with append because its harder to mess it up and as long as I call `ensureCapacityFor()` first it will be equivalent to a copy anyways.

Also thanks for pointing this out because I was forcing allocations by doing an append when I could use `ensureCapacityFor` to get a new byte slice from the pool if availabble